### PR TITLE
added security desensitization transform

### PR DESCRIPTION
**Pull request by wsyhj**
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-encrypt-phone/src/main/resources/META-INF/services/org.apache.seatunnel.spark.BaseSparkTransform [hunk 1]
```diff
@@ -0,0 +1 @@
+org.apache.seatunnel.spark.transform.EncryptedPhone
```
Comment (Hisoka-X) [inline seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-encrypt-phone/src/main/resources/META-INF/services/org.apache.seatunnel.spark.BaseSparkTransform:18]: Please add Apache License header, and add one empty end line.

**Comment by Hisoka-X** [comment]
Can you add doc to introduce this transform?

**Comment by wuchunfu** [comment]
@wsyhj Please resolve the conflicting files, thanks.

File: seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-encrypt-phone/src/main/resources/META-INF/services/org.apache.seatunnel.spark.BaseSparkTransform (hunk 1 above)
Comment (wsyhj) [reply seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-encrypt-phone/src/main/resources/META-INF/services/org.apache.seatunnel.spark.BaseSparkTransform:18]: Need to re request？

Comment (Hisoka-X) [reply seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-encrypt-phone/src/main/resources/META-INF/services/org.apache.seatunnel.spark.BaseSparkTransform:18]: Just push in `wsyhj:dev` branch

**Comment by wsyhj** [comment]
> @wsyhj Please resolve the conflicting files, thanks.

ok

**Comment by wsyhj** [comment]
> Can you add doc to introduce this transform?

refer to introduction.md

File: seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-encrypt-phone/introduction.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+### 目的
```
Comment (Hisoka-X) [inline seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-encrypt-phone/introduction.md]: Please use English to describe it.

**Comment by Hisoka-X** [comment]
Maybe after this PR merge, you can add one document in `docs/en/transform` directory. But should follow the template. After this, your Transform will show on our website.

File: seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-encrypt-phone/introduction.md (hunk 2 above)
Comment (wsyhj) [reply seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-encrypt-phone/introduction.md]: ok

**Comment by wsyhj** [comment]
> Maybe after this PR merge, you can add one document in `docs/en/transform` directory. But should follow the template. After this, your Transform will show on our website.

done

File: seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-encrypt-phone/src/main/resources/META-INF/services/org.apache.seatunnel.spark.BaseSparkTransform (hunk 1 above)
Comment (wsyhj) [reply seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-encrypt-phone/src/main/resources/META-INF/services/org.apache.seatunnel.spark.BaseSparkTransform:18]: done

**Review by Hisoka-X — APPROVED** [review]
LGTM, looking forward your next contribute!

**Comment by wsyhj** [comment]
> LGTM, looking forward your next contribute!

tks，but the request is suspend

**Review by CalvinKirs — COMMENTED** [review]
I don't think this is a general needs, or it is not suitable to be placed in SeaTunnel. Let's just look at it from a functional point of view. If you want to consider a phone number, or a mobile phone number in a non-Chinese region, then this function is not complete. to complete this function, many business scenarios need to be considered.

I don't recommend putting this kind of plug-in close to the business in the Plugin

**Comment by ruanwenjun** [comment]
I agree with CalvinKirs, it might be better to provide a general encrypt plugin rather than provide phone encrypt, id card encrypt...etc.

**Comment by wsyhj** [comment]
> I agree with CalvinKirs, it might be better to provide a general encrypt plugin rather than provide phone encrypt, id card encrypt...etc.

I got ,tks

**Pull request closed without merging**